### PR TITLE
[ru] Localization of /concepts/scheduling-eviction/assign-pod-node.md

### PR DESCRIPTION
**Pull request by kirkonru**
Add RU localization for assign-pod-node.md.

### CODE REVIEW

File: content/ru/docs/concepts/scheduling-eviction/assign-pod-node.md [hunk 1]
```diff
@@ -0,0 +1,409 @@
...
+2. Добавьте лейблы с префиксом `node-restriction.kubernetes.io/` к узлам и используйте их в [селекторах узлов](#nodeselector). Например, `example.com.node-restriction.kubernetes.io/fips=true` или `example.com.node-restriction.kubernetes.io/pci-dss=true`.
+
+## nodeSelector
+
+`nodeSelector` — простейшая рекомендуемая форма настроек выбора узлов.
+Можно добавить поле `nodeSelector` в спецификацию Pod'а и перечислить в нем [лейблы узлов](#built-in-node-labels), подходящих для развертывания Pod'а. В этом случае Kubernetes будет размещать Pod только на узлах с указанными лейблами.
+
+Дополнительную информацию см. в разделе [Размещение Pod'ов на узлах](/docs/tasks/configure-pod-container/assign-pods-nodes).
+
+## Правила совместного/раздельного существования (affinity и anti-affinity)
```
Comment (shurup) [inline content/ru/docs/concepts/scheduling-eviction/assign-pod-node.md:58]:
```suggestion
## Правила совместного/раздельного существования (affinity и anti-affinity) {#affinity-and-anti-affinity}
```
We need to keep the original titles here to make the anchor links work.

File: content/ru/docs/concepts/scheduling-eviction/assign-pod-node.md [hunk 2]
```diff
@@ -0,0 +1,409 @@
...
+  * Поле [nodeName](#nodename);
+  * [Ограничения на топологию распределения Pod'ов](#pod-topology-spread-constraints).
+
+## Лейблы узлов {#built-in-node-labels}
+
+Как и у многих других объектов Kubernetes, у узлов есть [лейблы](/docs/concepts/overview/working-with-objects/labels/). Их можно [навешивать вручную](/docs/tasks/configure-pod-container/assign-pods-nodes/#add-a-label-to-a-node).
+Kubernetes также предоставляет стандартный набор лейблов на всех узлах кластера. Список общих для узлов лейблов приведен в разделе [Типичные лейблы, аннотации и taint'ы](/docs/reference/labels-annotations-taints/).
+
+{{<note>}}
+Значения этих лейблов зависят от облачного провайдера и не гарантируются.
```
Comment (shurup) [inline content/ru/docs/concepts/scheduling-eviction/assign-pod-node.md:33]:
```suggestion
Значения этих лейблов зависят от облачного провайдера, поэтому нельзя точно на них полагаться.
```

File: content/ru/docs/concepts/scheduling-eviction/assign-pod-node.md [hunk 3]
```diff
@@ -0,0 +1,409 @@
...
+
+Лейблы узлов позволяют планировать Pod'ы на определенные узлы или группы узлов. С их помощью можно планировать Pod'ы на узлы с определенными требованиями к изоляции, безопасности или соответствию нормативным актам.
+
+При использовании лейблов для изоляции узлов следует выбирать ключи лейблов, которые {{<glossary_tooltip text="kubelet" term_id="kubelet">}} не может изменить. В этом случае взломанный узел не сможет навесить на себя эти лейблы в надежде заполучить рабочие нагрузки.
+
+[Admission-плагин NodeRestriction](/docs/reference/access-authn-authz/admission-controllers/#noderestriction) не позволяет kubelet'у устанавливать или изменять лейблы с префиксом `node-restriction.kubernetes.io/`.
+
+Чтобы использовать этот префикс для изоляции узла:
+
+
```
Comment (shurup) [inline content/ru/docs/concepts/scheduling-eviction/assign-pod-node.md:47]:
```suggestion
```

File: content/ru/docs/concepts/scheduling-eviction/assign-pod-node.md [hunk 4]
```diff
@@ -0,0 +1,409 @@
...
+## nodeSelector
+
+`nodeSelector` — простейшая рекомендуемая форма настроек выбора узлов.
+Можно добавить поле `nodeSelector` в спецификацию Pod'а и перечислить в нем [лейблы узлов](#built-in-node-labels), подходящих для развертывания Pod'а. В этом случае Kubernetes будет размещать Pod только на узлах с указанными лейблами.
+
+Дополнительную информацию см. в разделе [Размещение Pod'ов на узлах](/docs/tasks/configure-pod-container/assign-pods-nodes).
+
+## Правила совместного/раздельного существования (affinity и anti-affinity)
+
+`nodeSelector` — самый простой способ развернуть Pod'ы на узлах с определенными лейблами. Плавила совместного/раздельного существования расширяют типы ограничений, которые можно накладывать. Вот некоторые из их преимуществ:
```
Comment (shurup) [inline content/ru/docs/concepts/scheduling-eviction/assign-pod-node.md:60]:
```suggestion
`nodeSelector` — самый простой способ развернуть Pod'ы на узлах с определенными лейблами. Правила совместного/раздельного существования расширяют типы ограничений, которые можно накладывать. Вот некоторые из их преимуществ:
```

File: content/ru/docs/concepts/scheduling-eviction/assign-pod-node.md [hunk 5]
```diff
@@ -0,0 +1,409 @@
...
+* При планировании Pod'а планировщик может учитывать лейблы других Pod'ов, запущенных на узле (или в иной топологической области), а не только лейблы самого узла. Это позволяет формулировать правила, определяющие сосуществование Pod'ов на узле.
+
+Правила совместного существования (affinity) бывают двух типов:
+
+* *Правила для узлов* (node affinity) работают подобно полю `nodeSelector`, но более выразительны. Кроме того, можно задавать мягкие правила. 
+* *Правила для Pod'ов* позволяют при планировании учитывать наличие других Pod'ов (соответственно, и их лейблы) на узлах.
+
+### Правила совместного существования для узлов (node affinity)
+
+Правила совместного существования для узлов концептуально похожи на `nodeSelector`. С помощью лейблов они позволяют ограничивать список узлов, на которые может быть запланирован Существует два типа таких правил:
```
Comment (shurup) [inline content/ru/docs/concepts/scheduling-eviction/assign-pod-node.md:73]:
```suggestion
Правила совместного существования для узлов концептуально похожи на `nodeSelector`. С помощью лейблов они позволяют ограничивать список узлов, на которые может быть запланирован Pod. Существует два типа таких правил:
```

File: content/ru/docs/concepts/scheduling-eviction/assign-pod-node.md [hunk 6]
```diff
@@ -0,0 +1,409 @@
...
+В качестве примера рассмотрим следующую спецификацию Pod'а:
+
+{{< codenew file="pods/pod-with-node-affinity.yaml" >}}
+
+В этом примере применяются следующие правила:
+
+  * У узла *должен* быть лейбл с ключом `topology.kubernetes.io/zone`, и значение этого лейбла *должно* быть либо `antarctica-east1`, либо `antarctica-west1`.
+  * *Предпочтительно*, чтобы у узла был лейбл с ключом `another-node-label-key` и значением `another-node-label-value`.
+
+Можно использовать поле `operator` для указания логического оператора, который Kubernetes будет использовать при интерпретации правил. Доступны `In`, `NotIn`, `Exists`, `DoesNotExist`, `Gt` и `Lt`.
```
Comment (shurup) [inline content/ru/docs/concepts/scheduling-eviction/assign-pod-node.md:93]:
```suggestion
Можно использовать поле `operator` для указания логического оператора, который Kubernetes будет применять при интерпретации правил. Доступны `In`, `NotIn`, `Exists`, `DoesNotExist`, `Gt` и `Lt`.
```

File: content/ru/docs/concepts/scheduling-eviction/assign-pod-node.md [hunk 7]
```diff
@@ -0,0 +1,409 @@
...
+
+`NotIn` и `DoesNotExist` позволяют задавать правила раздельного существования (anti-affinity) для узла.
+Кроме того, можно использовать [taint'ы узлов](/docs/concepts/scheduling-eviction/taint-and-toleration/), чтобы "отвадить" Pod'ы от определенных узлов.
+
+{{<note>}}
+Когда указаны и `nodeSelector`, и `nodeAffinity`,  Pod будет запланирован на узел только в том случае, если *оба* этих параметра удовлетворены.
```
Comment (shurup) [inline content/ru/docs/concepts/scheduling-eviction/assign-pod-node.md:99]:
```suggestion
Когда указаны и `nodeSelector`, и `nodeAffinity`, Pod будет запланирован на узел только в том случае, если *оба* этих параметра удовлетворены.
```

File: content/ru/docs/concepts/scheduling-eviction/assign-pod-node.md [hunk 8]
```diff
@@ -0,0 +1,409 @@
...
+  
+Если указать несколько условий `nodeSelectorTerms`, привязанных к типам `nodeAffinity`, то Pod может быть запланирован на узел, если удовлетворяется одно из указанных условий `nodeSelectorTerms`.
+
+Если задать несколько выражений `matchExpressions` для одного условия `nodeSelectorTerms`, Pod будет запланирован на узел только если удовлетворены все выражения `matchExpressions`. {{}}
```
Comment (shurup) [inline content/ru/docs/concepts/scheduling-eviction/assign-pod-node.md:103]:
```suggestion
Если задать несколько выражений `matchExpressions` для одного условия `nodeSelectorTerms`, Pod будет запланирован на узел, только если удовлетворены все выражения `matchExpressions`. {{}}
```

File: content/ru/docs/concepts/scheduling-eviction/assign-pod-node.md [hunk 9]
```diff
@@ -0,0 +1,409 @@
...
+{{</note>}}
+
+Дополнительную информацию см. в разделе [Размещаем Pod'ы на узлы с помощью Node Affinity](/docs/tasks/configure-pod-container/assign-pods-nodes-using-node-affinity/).
+
+#### Вес правил совместного существования
+
+Для каждого правила типа `preferredDuringSchedulingIgnoredDuringExecution` можно указать вес `weight` в диапазоне от 1 до 100. Найдя несколько узлов, удовлетворяющих всем остальным требованиям для планирования Pod'а, планировщик перебирает предпочтительные правила, которым удовлетворяет узел, и прибавляет их вес для этого выражения к сумме.
+
+Итоговая сумма добавляется к оценке, полученной при анализе других параметров, влияющих на приоритета узл. Принимая решение о размещении Pod'а, планировщик отдает предпочтение узлам с наибольшей суммарной оценкой.
```
Comment (shurup) [inline content/ru/docs/concepts/scheduling-eviction/assign-pod-node.md:112]:
```suggestion
Итоговая сумма добавляется к оценке, полученной при анализе других параметров, влияющих на приоритет узла. Принимая решение о размещении Pod'а, планировщик отдает предпочтение узлам с наибольшей суммарной оценкой.
```

File: content/ru/docs/concepts/scheduling-eviction/assign-pod-node.md [hunk 10]
```diff
@@ -0,0 +1,409 @@
...
+            requiredDuringSchedulingIgnoredDuringExecution:
+              nodeSelectorTerms:
+              - matchExpressions:
+                - key: scheduler-profile
+                  operator: In
+                  values:
+                  - foo
+```
+
+Правило addedAffinity применяется ко всем Pod'ам с полем `.spec.schedulerName`, имеющем значение `foo-scheduler`, в дополнение к NodeAffinity, заданному в PodSpec. Таким образом, подходящие для Pod'а узлы должны удовлетворять параметрам `addedAffinity` и правилам `.spec.NodeAffinity` Pod'а.
```
Comment (shurup) [inline content/ru/docs/concepts/scheduling-eviction/assign-pod-node.md:150]:
```suggestion
Правило `addedAffinity` применяется ко всем Pod'ам с полем `.spec.schedulerName`, имеющем значение `foo-scheduler`, в дополнение к NodeAffinity, заданному в PodSpec. Таким образом, подходящие для Pod'а узлы должны удовлетворять параметрам `addedAffinity` и правилам `.spec.NodeAffinity` Pod'а.
```

**Review by shurup — CHANGES_REQUESTED** [review]
A great effort is made here. However, it's not fully translated yet. I reviewed the ready part of this document.

File: content/ru/docs/concepts/scheduling-eviction/assign-pod-node.md [hunk 11]
```diff
@@ -0,0 +1,349 @@
...
+  nodeName: kube-01
+```
+
+Такой под будет работать только на узле с именем `kube-01`.
+
+## Ограничения на топологию распределения подов.
+
+С помощью _ограничений на топологию распределения_ (_topology spread constraints_) можно настроить размещение {{< glossary_tooltip text="подов" term_id="Pod" >}} в кластере по failure-доменам, таким как регионы, зоны, узлы, или любым другим заданным топологическим конфигурациям. Это позволяет повысить производительность, ожидаемую доступность или эффективность использования ресурсов.
+
+Дополнительные сведения о принципах их работы читайте в разделе ["Ограничения на топологию распределения подов"](/docs/concepts/scheduling-eviction/topology-spread-constraints/).
```
Comment (kirkonru) [inline content/ru/docs/concepts/scheduling-eviction/assign-pod-node.md:340]:
```suggestion
Дополнительные сведения о принципах их работы читайте в разделе [Ограничения на топологию распределения подов](/docs/concepts/scheduling-eviction/topology-spread-constraints/).
```

File: content/ru/docs/concepts/scheduling-eviction/assign-pod-node.md [hunk 12]
```diff
@@ -0,0 +1,349 @@
...
+
+В принципе, `topologyKey` может быть любым разрешенным лейблом-ключом со следующими исключениями по соображениям производительности и безопасности:
+
+* При задании правил совместного/раздельного существования для подов пустое поле `topologyKey` не допускается как для `requiredDuringSchedulingIgnoredDuringExecution`, так и для `preferredDuringSchedulingIgnoredDuringExecution`.
+* Для правил раздельного существования типа `requiredDuringSchedulingIgnoredDuringExecution` admission-контроллер разрешает использовать только `kubernetes.io/hostname` в качестве `topologyKey`. Для работы с кастомными топологиями admission-контроллер можно дополнить или совсем отключить его.
+
+В дополнение к `labelSelector` и `topologyKey` можно опционально указать список пространств имен, которые должен выбирать `labelSelector`, с помощью поля `namespaces` на том же уровне, что и `labelSelector` и `topologyKey`. Если поле `namespaces` опущено или пусто, по умолчанию выбирается пространство имен пода, в котором задаются правила совместного/раздельного существования.
```
Comment (shurup) [inline content/ru/docs/concepts/scheduling-eviction/assign-pod-node.md:208]:
```suggestion
В дополнение к `labelSelector` и `topologyKey` можно опционально указать список пространств имен, которые должен выбирать `labelSelector`, с помощью поля `namespaces` на том же уровне, что `labelSelector` и `topologyKey`. Если поле `namespaces` опущено или пусто, по умолчанию выбирается пространство имен пода, в котором задаются правила совместного/раздельного существования.
```

File: content/ru/docs/concepts/scheduling-eviction/assign-pod-node.md [hunk 13]
```diff
@@ -0,0 +1,349 @@
...
+#### Селектор пространств имен
+{{< feature-state for_k8s_version="v1.24" state="stable" >}}
+
+Подходящие пространства имен также можно выбрать с помощью `namespaceSelector`, который попытается найти лейбл в наборе пространств имен.
+Условия совместного существования применяются к пространствам имен, выбранным как селектором `namespaceSelector`, так и полем `namespaces`.
+Обратите внимание, что пустой селектор `namespaceSelector` ({}) выбирает все пространства имен, в то время как пустой или null-список `namespaces` и null-селектор `namespaceSelector` выбирает пространство имен пода, в котором правило задано.
+
+#### Другие примеры использования
+
+Правила совместного/раздельного существования для подов особенно удобны, когда используются совместно с абстракциями более высокого уровня, такими как ReplicaSet, StatefulSet, Deployment, и т.д. Эти правила позволяют настроить размещение рабочих нагрузок с учетом имеющейся топологии; например, пара связанных подов будет размещена на одном узле.
```
Comment (shurup) [inline content/ru/docs/concepts/scheduling-eviction/assign-pod-node.md:219]:
```suggestion
Правила совместного/раздельного существования для подов особенно удобны, когда используются совместно с абстракциями более высокого уровня, такими как ReplicaSet, StatefulSet, Deployment и т.д. Эти правила позволяют настроить размещение рабочих нагрузок с учетом имеющейся топологии; например, пара связанных подов будет размещена на одном узле.
```

File: content/ru/docs/concepts/scheduling-eviction/assign-pod-node.md [hunk 14]
```diff
@@ -0,0 +1,349 @@
...
+                operator: In
+                values:
+                - store
+            topologyKey: "kubernetes.io/hostname"
+      containers:
+      - name: web-app
+        image: nginx:1.16-alpine
+```
+
+Развертывание ресурсов в соответствии с приведенными выше конфигурациями приведет к созданию кластера, в котором на каждом узле будет по одному веб-серверу и одное реплике Redis (всего три отдельных узла):
```
Comment (shurup) [inline content/ru/docs/concepts/scheduling-eviction/assign-pod-node.md:297]:
```suggestion
Развертывание ресурсов в соответствии с приведенными выше конфигурациями приведет к созданию кластера, в котором на каждом узле будет по одному веб-серверу и одной реплике Redis (всего три отдельных узла):
```

File: content/ru/docs/concepts/scheduling-eviction/assign-pod-node.md [hunk 15]
```diff
@@ -0,0 +1,349 @@
...
+spec:
+  containers:
+  - name: nginx
+    image: nginx
+  nodeName: kube-01
+```
+
+Такой под будет работать только на узле с именем `kube-01`.
+
+## Ограничения на топологию распределения подов.
```
Comment (shurup) [inline content/ru/docs/concepts/scheduling-eviction/assign-pod-node.md:336]:
```suggestion
## Ограничения на топологию распределения подов
```

**Review by shurup — CHANGES_REQUESTED** [review]
Thank you, @kirkonru! Here are a few tiny corrections for me.

More importantly, I think the `content/ru/docs/reference/issues-security/*` files were included in this PR by some mistake? They are not translated and don't seem to be relevant to the main article.

**Comment by divya-mohan0209** [comment]
Hello @kirkonru , please do not make changes to multiple localization files over the same PR since the process for each localization differs.

**Comment by tengqm** [comment]
/close
A long time has passed.
Feel free to reopen this PR to move it forward.

**Pull request closed without merging**